line_profiler/_line_profiler.pyx

```diff
@@ -156,72 +203,75 @@ cdef class LineProfiler:
 
     def enable(self):
         PyEval_SetTrace(python_trace_callback, self)
-
-    def disable(self):
-        self.last_time = {}
+        
+    @property
+    def code_map(self):
+        return <dict>self.c_code_map
+        
+    @property
+    def last_time(self):
+        return <dict>self.c_last_time
+
+    cpdef disable(self):
+        self.c_last_time.clear()
         unset_trace()
 
-    def get_stats(self):
+    cpdef get_stats(self):
         """ Return a LineStats object containing the timings.
         """
+        cdef dict cmap
+        
         stats = {}
-        for code in self.code_map:
-            entries = self.code_map[code].values()
+        for code in self.code_hash_map:
+            cmap = self.c_code_map
```

Comment:
This works even though cmap is a `dict` and `c_code_map` is an `unordered_map`?

Reply:
Yep! Cython can infer types :)